### ISSUE TITLE
[Question] How to remove window decorations?

### ISSUE THREAD

**Issue by kanyck** [WayfireWM/wayfire#551]
Fiddling with wayfire for some time and quite impressed. I've been a compiz user for over a decade and watching sadly how it was gradually dying... So I'm happy that wayfire shows up. 
However.
I want to remove title bar from native wayland windows (at least from Alacritty's, maybe from all). Because wayfire allows to move/resize windows without using title bar, it looks like it doesn't carry any useful functionality anymore and just wastes space (especially on a small screens). 
Switching off decoration module or setting title bar height to 0 doesn't do anything visible.
I even can't understand who is responsible for, say, Alacritty decorations, because setting them off in alacritty config file has no effect, either.

**Comment by ammen99** [comment]
One hack you can try is setting the preferred decoration mode to server via https://github.com/WayfireWM/wayfire/wiki/Configuration#corepreferred_decoration_mode

This would make some of the apps switch to using wayfire's decorations, which are actually not there if you disable the decoration plugin.

This however won't work with GTK apps and possibly many others - you'll have to configure them via toolkit specific options. Gtk has some modules to disable headerbars, etc.

**Comment by kanyck** [comment]
Well, it doesn't help with `alacritty`. Decorations are still there even when they're switched off on both sides. 
However, `wcm` crashes wayfire with `preferred_decoration_mode = server`. Whats even worse, the wayfire crash causes video driver to get stuck, so I may either use kernel magic Alt+SysRq+R... to reboot or ssh from another machine (to reboot, yes). I probably should file another bug about it. Please suggest what kind of debug info may I provide to you for investigation, given that screen is unusable after crash, and `tee`-ing debug log to a file will probably loose last, the most important part of it due to bufferization.

**Comment by ammen99** [comment]
Does this happen if wayfire is running nested inside another compositor? If that is the case, then most probably your system won't crash if you can reproduce the crash there and you'll be able to generate a stack trace.

Also, if you have a bug which you can reliably reproduce, please compile wayfire with ASAN. While wayfire prints a stack trace by default, ASAN can detect subtler memory corruption issues so it is better for reports.

**Comment by kanyck** [comment]
Okay, I'll try to investigate it further. (first I want to eventually finish building firefox. It's probably the 4-th start over). I already experienced wayfire random crashes a couple of times, that's probably the chance to catch up.

**Comment by elenapan** [comment]
> Well, it doesn't help with `alacritty`. Decorations are still there even when they're switched off on both sides.

Same here. This has to be an alacritty bug. I managed to get rid of them by setting `decorations: full` in the alacritty config and `preferred_decoration_mode = server` in the wayfire config.

**Comment by kanyck** [comment]
There's thing like https://github.com/PCMan/gtk3-nocsd
that's supposed to partially solve the problem. However, the root problem seems to be RedHat and their policies...

**Comment by ammen99** [comment]
> Same here. This has to be an alacritty bug. I managed to get rid of them by setting `decorations: full` in the alacritty config and `preferred_decoration_mode = server` in the wayfire config.

Oh, so my guess is, if you tell alacritty to have only partial decorations, it ignores the compositor's preference for server-side decorations (because only the app can have partial decors). Nothing we can do here.




> There's thing like https://github.com/PCMan/gtk3-nocsd
> that's supposed to partially solve the problem. However, the root problem seems to be RedHat and their policies...

Yeah, there's not that much Wayfire can do, if GTK doesn't respect the xdg-decoration protocol.


Anyway, I'll close this issue because there isn't much we can do on our side. Looking forward to your crash reports in a new issue @kanyck  :)

**Comment by TRPB** [comment]
I realised this is closed but can anyone help me achieve this with sublime text? 

In KDE I can set `"gtk_client_side_window_decorations": false,` in sublime's options, then disable the border in window rules (or right click the taskbar icon, more actions > no border) so it's certainly possible.

In theory, I should be able to set `gtk_client_side_window_decorations` to false and disable it using the compositor or to true and disable it with libgtk3-nocsd.so, but neither option seem to work. 

To disable decorations I've set  this in wayfire.ini, which certainly works for native wayland applications:

```
[decoration]
border_size = 0
title_height = 0
ignore_views = all
```
